lib/bolt/config.rb

```diff
@@ -189,15 +189,13 @@ def initialize(project, config_data, overrides = {})
 
       # If we need to lower concurrency and concurrency is not configured
       ld_concurrency = loaded_data.map(&:keys).flatten.include?('concurrency')
-      if default_concurrency != DEFAULT_DEFAULT_CONCURRENCY &&
-         !ld_concurrency &&
-         !override_data.key?('concurrency')
-        concurrency_warning = { option: 'concurrency',
-                                msg: "Concurrency will default to #{default_concurrency} because ulimit "\
-                                "is low: #{Etc.sysconf(Etc::SC_OPEN_MAX)}. Set concurrency with "\
-                                "'--concurrency', or set your ulimit with 'ulimit -n <limit>'" }
-        @warnings << concurrency_warning
-      end
+      @modified_concurrency = if default_concurrency != DEFAULT_DEFAULT_CONCURRENCY &&
+                                 !ld_concurrency &&
+                                 !override_data.key?('concurrency')
+                                true
+                              else
+                                false
+                              end
```

Comment:
This can just directly assign the boolean expression instead of having an if/else.